tldw_Server_API/app/core/AuthNZ/repos/shared_workspace_repo.py

```
@@ -0,0 +1,541 @@
+            for row in rows
+            if (name := self._row_to_dict(row).get("name")) and str(name) in required
+        }
+        missing = required - existing
+        if missing:
+            raise RuntimeError(
+                "Sharing tables are missing. Run AuthNZ migrations. "
+                f"Missing: {sorted(missing)}"
+            )
+
+    @staticmethod
+    def _row_to_dict(row: Any) -> dict[str, Any]:
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `row`

(ANN401)